spec/core/complex/to_s_spec.rb

```diff
@@ -41,4 +42,13 @@
   it "returns 1+NaN*i for Complex(1, NaN)" do
     Complex(1, nan_value).to_s.should == "1+NaN*i"
   end
+
+  it "treats real and imaginary parts as strings" do
+    real = NumericSpecs::Subclass.new
+    real.should_receive(:to_s).and_return("1")
+    imaginary = NumericSpecs::Subclass.new
+    imaginary.should_receive(:to_s).and_return("2")
+    imaginary.should_receive(:<).any_number_of_times.and_return(false)
+    Complex(real, imaginary).to_s.should == "1+2i"
+  end
```

Comment:
We need to either upstream this to [ruby/spec](https://github.com/ruby/spec) or move it to our own `test/natalie/complex_test.rb`, otherwise it will be lost when we update specs from ruby/spec later.

Reply:
It's from ruby/spec, I updated from here:
https://github.com/ruby/spec/blob/master/core/complex/to_s_spec.rb#L46-L53

Reply:
Oh, my bad. I checked a local copy of ruby/spec and failed to `git pull`. Thanks!